src/sort-scripts.ts

```diff
@@ -3,31 +3,57 @@ import orderBy from 'sort-order';
 import { PackageJson } from './types';
 
 const PRE_OR_POST_PREFIX = /^(pre|post)/;
+const NPM_RUN_ALL_SEPARATOR = /([:/])/;
+const NPM_BUILTIN_SCRIPTS = ['install', 'prepare', 'pack', 'publish', 'restart', 'start', 'stop', 'test', 'version', 'uninstall'];
+
+function parseScriptName(scripts: PackageJson['scripts'], script: string): { prefix: '' | 'pre' | 'post'; name: string } {
+  if (script === 'prepublishOnly') return { prefix: 'pre', name: 'publish' };
+  const prefixMatch = PRE_OR_POST_PREFIX.exec(script);
+  const prefix = prefixMatch ? prefixMatch[0] as 'pre' | 'post' : '';
+  const name = script.slice(prefix.length);
+  if (scripts!.hasOwnProperty(name) || NPM_BUILTIN_SCRIPTS.includes(name)) return { prefix, name };
+  return { prefix: '', name: script };
+}
 
 // Sort alphabetically by script name excluding pre/post prefixes
-function scriptName(...args: [string, string]): 1 | 0 | -1 {
-  const [a, b] = args.map((arg) => arg.replace(PRE_OR_POST_PREFIX, ''));
-  if (a !== b) {
-    return a < b ? -1 : 1;
-  } else {
+function scriptName(scripts: PackageJson['scripts']) {
+  return function (a: string, b: string): 1 | 0 | -1 {
+    const an = parseScriptName(scripts, a).name;
+    const bn = parseScriptName(scripts, b).name;
+    const aa = an.split(NPM_RUN_ALL_SEPARATOR);
+    const bb = bn.split(NPM_RUN_ALL_SEPARATOR);
+    const al = aa.length;
+    const bl = bb.length;
+    for (let i = 0; i < al || i < bl; i++) {
```

Comment:
**Suggestion:** We can remove the need for `al` and `bl` using `Math.min`.

```ts
for (let i = 0, len = Math.min(aa.length, bb.length); i < len; i++) {
```

Reply:
I cannot make this change because `al` and `bl` are also used on line 32.

However, I realize that the problem is with the short variable names that I chose. I will also fix this issue.